src/stdune/dyn0.ml

```diff
@@ -19,96 +22,147 @@ type t =
   | Map of (t * t) list
   | Set of t list
 
-let rec pp = function
-  | Opaque -> Pp.verbatim "<opaque>"
-  | Unit -> Pp.verbatim "()"
-  | Int i -> Pp.verbatim (string_of_int i)
-  | Bool b -> Pp.verbatim (string_of_bool b)
-  | String s -> Pp.verbatim s
-  | Bytes b -> Pp.verbatim (Bytes.to_string b)
-  | Char c -> Pp.char c
-  | Float f -> Pp.verbatim (string_of_float f)
-  | Sexp s -> pp_sexp s
-  | Option None -> pp (Variant ("None", []))
-  | Option (Some x) -> pp (Variant ("Some", [x]))
-  | List x ->
-    Pp.box
-      (Pp.concat
-         [ Pp.char '['
-         ; Pp.concat_map ~sep:(Pp.seq (Pp.char ';') Pp.space) x ~f:pp
-         ; Pp.char ']'
-         ])
-  | Array a ->
-    Pp.box
-      (Pp.concat
-         [ Pp.verbatim "[|"
-         ; Pp.concat_map ~sep:(Pp.seq (Pp.char ';') Pp.space) (Array.to_list a) ~f:pp
-         ; Pp.verbatim "|]"
-         ])
-  | Set xs ->
-    Pp.box
-      (Pp.concat
-         [ Pp.verbatim "set {"
-         ; Pp.concat_map ~sep:(Pp.seq (Pp.char ';') Pp.space) xs ~f:pp
-         ; Pp.verbatim "}"
-         ])
-  | Map xs ->
-    Pp.box
-      (Pp.concat
-         [ Pp.verbatim "map {"
-         ; Pp.concat_map ~sep:(Pp.seq (Pp.char ';') Pp.space) xs ~f:(fun (k, v) ->
-             Pp.box
-               (Pp.concat
-                  [ pp k
-                  ; Pp.space
-                  ; Pp.verbatim ":"
-                  ; Pp.space
-                  ; pp v
-                  ])
-           )
-         ; Pp.verbatim "}"
-         ])
-  | Tuple x ->
-    Pp.box
-      (Pp.concat
-         [ Pp.char '('
-         ; Pp.concat_map ~sep:(Pp.seq (Pp.char ',') Pp.space) x ~f:pp
-         ; Pp.char ')'
-         ])
-  | Record fields ->
-    Pp.vbox ~indent:2
-      (Pp.concat
-         [ Pp.char '{'
-         ; Pp.concat_map ~sep:(Pp.char ';') fields ~f:(fun (f, v) ->
-             Pp.concat
-               [ Pp.verbatim f
-               ; Pp.space
-               ; Pp.char '='
-               ; Pp.space
-               ; Pp.box ~indent:2 (pp v)
-               ]
-           )
-         ; Pp.char '}'
-         ])
-  | Variant (v, []) -> Pp.verbatim v
-  | Variant (v, xs) ->
-    Pp.hvbox ~indent:2
-      (Pp.concat
-         [ Pp.verbatim v
-         ; Pp.space
-         ; Pp.concat_map ~sep:(Pp.char ',') xs ~f:pp
-         ])
+  let unsnoc l = match List.rev l with
+    | last :: before_last -> Some (List.rev before_last, last)
+    | [] -> None
+
+  let split s ~on =
```

Comment:
could you move this function to the `string_split` module? that one should have no dependencies.

Reply:
Uh, OK, I'll see what can be done.

Reply:
Oh well, this was trivial, I just did what you said. Thanks!